src/gui/qgsexpressionbuilderwidget.cpp

```diff
@@ -443,59 +466,112 @@ void QgsExpressionBuilderWidget::newFunctionFile( const QString &fileName )
 
 void QgsExpressionBuilderWidget::btnNewFile_pressed()
 {
-  bool ok;
-  QString text = QInputDialog::getText( this, tr( "New File" ),
-                                        tr( "New file name:" ), QLineEdit::Normal,
-                                        QString(), &ok );
-  if ( ok && !text.isEmpty() )
+  // If a project has an entry for functions, then we should
+  // already have a 'Project functions' item in the file list.
+  // Since only one item should correspond to 'Project functions',
+  // we'll disable this option in the 'add function file' dialog.
+  bool ok = false;
+  mProject->readEntry( QStringLiteral( "ExpressionFunctions" ), QStringLiteral( "/pythonCode" ), QString(), &ok );
+
+  QgsExpressionAddFunctionFileDialog dlg {!ok, this};
+  if ( dlg.exec() == QDialog::DialogCode::Accepted )
   {
-    newFunctionFile( text );
+    if ( dlg.createProjectFunctions() )
+    {
+      QListWidgetItem *item = new QListWidgetItem( QgsApplication::getThemeIcon( QStringLiteral( "mIconQgsProjectFile.svg" ) ), DEFAULT_PROJECT_FUNCTIONS_ITEM_NAME );
+      item->setData( Qt::UserRole, QStringLiteral( "project" ) );
+      cmbFileNames->insertItem( 0, item );
+      cmbFileNames->setCurrentRow( 0 );
+
+      QString templatetxt;
```

Comment:
I'd avoid acronyms here, expanded wording == easier to read. I.e. here, templateText > templateTxt :)

Reply:
Done!